platform/android/export/export.cpp

```diff
@@ -2625,35 +2660,58 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 				EditorNode::get_singleton()->show_warning(TTR("Building of Android project failed, check output for the error.\nAlternatively visit docs.godotengine.org for Android build documentation."));
 				return ERR_CANT_CREATE;
 			}
-			if (p_debug) {
-				src_apk = build_path.plus_file("build/outputs/apk/debug/android_debug.apk");
-			} else {
-				src_apk = build_path.plus_file("build/outputs/apk/release/android_release.apk");
+
+			List<String> copy_args;
+			String copy_command;
+			if (export_format == 1) {
+				copy_command = vformat("copyAndRename%sAab", build_type);
+			} else if (export_format == 0) {
+				copy_command = vformat("copyAndRename%sApk", build_type);
 			}
 
-			if (!FileAccess::exists(src_apk)) {
-				EditorNode::get_singleton()->show_warning(TTR("No build apk generated at: ") + "\n" + src_apk);
+			copy_args.push_back(copy_command);
+
+			copy_args.push_back("-p"); // argument to specify the start directory.
+			copy_args.push_back(build_path); // start directory.
+
+			String export_filename = p_path.get_file();
+			String export_path = p_path.get_base_dir().replace("res://", "../../");
+
+			copy_args.push_back("-Pexport_path=" + export_path);
+			copy_args.push_back("-Pexport_filename=" + export_filename);
+
+			int copy_result = EditorNode::get_singleton()->execute_and_show_output(TTR("Moving output"), build_command, copy_args);
+			if (copy_result != 0) {
+				EditorNode::get_singleton()->show_warning(TTR("Unable to copy and rename export file, check gradle project directory for outputs."));
 				return ERR_CANT_CREATE;
 			}
 
-		} else {
+			if (export_format == 0) {
+				if (_signed) {
+					err = sign_apk(p_preset, p_debug, p_path, ep);
+					if (err != OK) {
+						return err;
+					}
+				}
+			}
+			return OK;
+		}
 
-			if (p_debug)
-				src_apk = p_preset->get("custom_template/debug");
-			else
-				src_apk = p_preset->get("custom_template/release");
+		if (p_debug)
```

Comment:
Can you add a comment above this line to denote this is the start of the legacy build path.